game/tests.py

```
@@ -1474,6 +1489,253 @@ def test_promotion_notation_invalid_piece_defaults_to_queen(self):
         self.assertEqual(notation, 'a8=Q')
 
 
+class SecureRegistrationTest(TestCase):
+    """Security-focused tests for the hardened registration flow."""
+
+    VALID_PAYLOAD = {
+        'username': 'newchessplayer',
+        'email': 'newchessplayer@example.com',
+        'password1': 'StrongPass123!',
+        'password2': 'StrongPass123!',
+    }
```

Ruff (0.15.15)

[warning] Mutable default value for class attribute

(RUF012)